### ISSUE TITLE
Question: Is it possible to store result of callbacks e.g. at end of each Talos run?

### ISSUE THREAD

**Issue by bml1g12** [autonomio/talos#81]
Question: Is it possible to store result of callbacks e.g. at end of each Talos run? Or any other such arbitrary python code. 

I'd like to write a custom metric which involves some quite complicated operations and simply appending the result of arbitrary python code at the end (or during) a talos parameter-iteration would be very handy.  

I started trying to write a Keras metric to do what I want, but its a real pain as I need to operate on tensors with  (?, ?, ?) shape (undefined on compile time) and thus can't do things like calculate the length of the array, or use for loops. i.e. the code has to be properly vectorized.

Edit: In fact, if there is a way to simply save the keras model from the each talos parameter-iteration, then that would also let me run arbitrary code. Does "save_best_model" still exist?

**Comment by mikkokotila** [comment]
Here is the discussion on Keras issues related with this [[link]](https://github.com/keras-team/keras/issues/2397#issuecomment-416733633)

**Comment by mikkokotila** [comment]
To clarify, I don't know about the other properties of the model object, but I know that there are issues regarding making predictions with a saved model when the models are created in an iterative process (such as hyperparameter scanning).

**Comment by bml1g12** [comment]
Hi @mikkokotila  thanks for your help.  I guess it is not possible currently then. I'll close this question - saving the model or allowing arbitrary callbacks would be a nice feature though, as users might wish to run calculate criteria for model selection which are not easily computed as Keras metrics.
